File: find_dashes.py
import os

def find_dashes(directory):
    em_dash = '—'
    en_dash = '–'
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(('.ts', '.tsx', '.js', '.jsx', '.json')):
                path = os.path.join(root, file)
                try:
                    with open(path, 'r', encoding='utf-8') as f:
                        lines = f.readlines()
                        for i, line in enumerate(lines):
                            if em_dash in line or en_dash in line:
                                print(f"File: {path}, Line: {i+1}")
                                if em_dash in line:
                                    print(f"  Found em-dash (—)")
                                if en_dash in line:
                                    print(f"  Found en-dash (–)")
                                print(f"  Content: {line.strip()}")
                except Exception as e:
                    print(f"Error reading {path}: {e}")

if __name__ == "__main__":
    find_dashes('src')

File: find_dashes.js
const fs = require('fs');
const path = require('path');

function findDashes(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        if (stat.isDirectory()) {
            if (file !== 'node_modules' && file !== '.next' && file !== 'dist') {
                findDashes(fullPath);
            }
        } else if (file.endsWith('.ts') || file.endsWith('.tsx') || file.endsWith('.js') || file.endsWith('.jsx') || file.endsWith('.json')) {
            const content = fs.readFileSync(fullPath, 'utf8');
            const lines = content.split('\n');
            lines.forEach((line, index) => {
                if (line.includes('—') || line.includes('–')) {
                    console.log(`File: ${fullPath}, Line: ${index + 1}`);
                    if (line.includes('—')) console.log('  Found em-dash (—)');
                    if (line.includes('–')) console.log('  Found en-dash (–)');
                    console.log(`  Content: ${line.trim()}`);
                }
            });
        }
    }
}

findDashes('src');
